src/Tools/Directory.Build.targets

```diff
@@ -8,7 +8,7 @@
       Target the default version of .NET Core in tool projects to maximize the compatible environments. Must be set
       before importing root Directory.Build.targets.
     -->
-    <TargetLatestDotNetRuntime Condition=" '$(IsServicingBuild)' == 'true' ">false</TargetLatestDotNetRuntime>
+    <TargetLatestDotNetRuntime Condition=" '$(IsServicingBuild)' == 'true' '$(DotNetSourceBuild))' != 'true'">false</TargetLatestDotNetRuntime>
```

Comment:
This runs counter to the "maximize the compatible environments" comment just above. I suspect it's not a horrible change (modulo the missing `and`) for the tools that ship w/in the SDK. In fact, it's probably the right thing to do for all such tools even outside source build.

But it isn't correct for `dotnet-sql-cache` and most other tools. They use this setting to ensure they remain compatible w/ the first of any major release. Please instead either make this conditional on the specific projects that should be affected or move the setting down into the individual projects (for all but the three packages listed in the PR description).

Reply:
I pushed a change that just copies the condition - let me know if you'd prefer to remove the SourceBuild conditional on this.  Thanks!